crate/cli/src/actions/cover_crypt/encrypt.rs

```diff
@@ -82,18 +102,151 @@ impl EncryptAction {
+
+    /// Several files need to be encrypted.
+    /// A custom protocol is used to serialize these data.
+    ///
+    /// Bulk encryption / decryption scheme
+    ///
+    /// ENC request
+    /// | nb_chunks (LEB128) | chunk_size (LEB128) | chunk_data (plaintext)
+    ///                        <------------- nb_chunks times ------------>
+    ///
+    /// ENC response
+    /// | EH | nb_chunks (LEB128) | chunk_size (LEB128) | chunk_data (encrypted)
+    ///                             <------------- nb_chunks times ------------>
+    ///
+    /// DEC request
+    /// | nb_chunks (LEB128) | size(EH + chunk_data) (LEB128) | EH | chunk_data (encrypted)
+    ///                                                         <----- chunk with EH ----->
+    ///                        <---------------------- nb_chunks times ------------------->
+    ///
+    /// DEC response
+    /// | nb_chunks (LEB128) | chunk_size (LEB128) | chunk_data (plaintext)
+    ///                        <------------- nb_chunks times ------------>
+    fn prepare_data_bulk(&self) -> Result<Vec<u8>, CliError> {
+        let mut data = Vec::new();
+
+        // number of encrypted files
+        leb128::write::unsigned(&mut data, self.input_files.len() as u64).map_err(|_| {
+            CliError::Conversion("Cannot write the number of files to encrypt".to_string())
+        })?;
+
+        for input_file in &self.input_files {
+            // Read the file to encrypt
+            let mut f = File::open(input_file)
+                .with_context(|| format!("Can't read the file to encrypt ({input_file:?})"))?;
+            let mut content = Vec::new();
+            f.read_to_end(&mut content)
+                .with_context(|| "Fail to read the file to encrypt")?;
+
+            leb128::write::unsigned(&mut data, content.len() as u64).map_err(|_| {
+                CliError::Conversion("Cannot write the size of the chunk to encrypt".to_string())
+            })?;
+            data.append(&mut content);
+        }
+
+        Ok(data)
+    }
+
+    /// Read the bulk serialized data in order to write it on disk.
+    /// Each encrypted chunk is written starting with the encrypted header,
+    /// so each chunk can be decrypted independently.
+    fn write_bulk_encrypted_data(&self, data: &[u8]) -> Result<(), CliError> {
+        // Read encrypted header
+        let mut de = Deserializer::new(data);
+        let encrypted_header = EncryptedHeader::read(&mut de)
+            .map_err(|_| {
+                CliError::Conversion(
+                    "Unable to recognize encrypted header structure from slice".to_string(),
+                )
+            })?
+            .serialize()
+            .map_err(|_| {
+                CliError::Conversion(
+                    "Unable to serialize encrypted header structure to bytes".to_string(),
+                )
+            })?;
+        let encrypted_block = de.finalize();
+
+        let mut data_slice: &[u8] = encrypted_block.as_ref();
```

Comment:
Why to create this variable?

Reply:
the prototype of `::core::slice::take` require the use of a dedicated `mut _: &[u8]` variable